Add kube-slack chart

jstriebel opened the pull request:
This PR adds an incubator chart for [`kube-slack`](https://github.com/wongnai/kube-slack).

I tried to incorporate best practices as far as possible. I did not enter any `maintainers` in the `Chart.yaml` yet. Should I enter myself there?

If all requirements are met, I can also move it to `stable`.

Thanks for considering this PR!

/cc @whs (author of `kube-slack`)

jstriebel committed:
add kube-slack chart

jstriebel committed:
Merge master into add-kube-slack

jstriebel committed:
kube-slack: use generic rbac apiVersion template

jstriebel committed:
kube-slack improve documentation

jstriebel commented:
/assign @foxish

jstriebel committed:
kube-slack: update NOTES

foxish commented:
> I tried to incorporate best practices as far as possible. I did not enter any maintainers in the Chart.yaml yet. Should I enter myself there?

Yes, that would be a good idea.

foxish commented:
/ok-to-test

jstriebel committed:
kube-slack: add jstriebel as maintainer

jstriebel commented:
@foxish done, added myself as maintainer. Is anything missing to move this to `stable`?

jstriebel commented:
@foxish anything else I should do about this chart?

rawkode commented:
Would love to see this merged 👍

unguiculus commented:
/hold

unguiculus commented:
/assign

unguiculus commented on incubator/kube-slack/templates/_helpers.tpl:
```diff
@@ -0,0 +1,27 @@
+Return the appropriate apiVersion for RBAC APIs.
+*/}}
+{{- define "rbac.apiVersion" -}}
+{{- if ge .Capabilities.KubeVersion.Minor "8" -}}
```
This will not work once Kubernetes 1.10 is out (8 > 10). You'd have to use `semvercompare`. However, I'd suggest you just use the `v1` api or do you absolutely have to support K8s < 1.8? Officially, we support the current and previous major versions, which would be 1.8+.

unguiculus reviewed (changes requested):
Thanks for the PR. I'd like you to do a few corrections. Please compare your templates with what you get with `helm create` using Helm 2.8+ and update accordingly.

Please follow our RBAC best practices.
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

rawkode commented:
@jstriebel Are you able to make these changes, or can I pick them up?

jstriebel commented:
Sorry for the late reply, I was unavailable the last week. @rawkode, I would add the changes the next days, but feel free to pick it up.

jstriebel committed:
adopt new helm best practices

jstriebel committed:
fix rbac.apiVersion comparison for future releases

jstriebel commented:
@unguiculus Thank you for the feedback, I changed the chart accordingly.

As I would still like to use this chart with kubernetes 1.7, I used `semverCompare`. The linter had a bug about the usage of , which [is already fixed](https://github.com/kubernetes/helm/issues/3349), only the [testing needs to be upgraded to the newest helm version](https://github.com/kubernetes/charts/blob/master/test/circle/install.sh#L17).

To support kubernetes 1.7, I used `apps/v1beta1` as the apiVersion for the deployment, I hope that's fine. I already use the explicit selector which is needed in future versions.

If you think that is to much clutter for the compatibility, I would change the rbac and deployment versions accordingly.

jstriebel committed:
add notReadyMinTime value

jstriebel committed:
fix whitespace, obey the linter

jstriebel committed:
quote notReadyMinTime

jstriebel committed:
trigger CI

jstriebel committed:
Merge remote-tracking branch 'kube/master' into add-kube-slack

jstriebel committed:
fix Chart.yaml syntax, add home

jstriebel commented:
As the newest helm version is now used, I guess this should be ready. Could you have another look, @unguiculus?

/assign @unguiculus

unguiculus commented on incubator/kube-slack/templates/_helpers.tpl:
```diff
@@ -0,0 +1,43 @@
+Return the appropriate apiVersion for RBAC APIs.
+*/}}
+{{- define "rbac.apiVersion" -}}
+{{- if semverCompare ">= 1.8" .Capabilities.KubeVersion.GitVersion -}}
```
In order to be compatible with e. g. GKE, which names its release as pre-releases, please do this:

`if semverCompare ">= 1.8-0"`

See #3002

unguiculus reviewed (changes requested):
Let me remind you to apply our RBAC best practices:
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Also, I'd suggest you directly move the chart to stable.

jstriebel committed:
incorporate PR feedback

jstriebel committed:
move to stable

jstriebel committed:
clean up rbac values descriptions

jstriebel commented:
@unguiculus Thanks for the feedback! I incorporated it and moved the chart to stable. I also renamed the `rbac.apiVersion` helper to `kube-slack.rbac.apiVersion`. Is anything else missing?

PS: Sorry for the delay.

unguiculus commented:
/lgtm

unguiculus commented:
/hold cancel
